### PR TITLE
Add Cumulative Weighting to the Wheel Algo

### PR DESCRIPTION
**Pull request by coreyja**
Now there is a bit more randomness about which Restaurants will
be on any given wheel!
No longer deterministic, so gonna delete some specs :lol:

### CODE REVIEW

**Review by danf1024 — APPROVED** [review]
lgtm

File: app/models/restaurant_wheel.rb [hunk 1]
```diff
@@ -30,17 +40,22 @@ def filtered_restaurants
...
+      [total, restaurant]
     end
-    @wheel = @wheel.first(num_stops).shuffle
+  end
+
+  def choose_single_restaurant
+    number = rand
+    cummulative_weighted_restaurants.detect do |cummulative_weighting, _|
+      cummulative_weighting >= number
+    end&.last
```
Comment (LucasSantoro) [inline app/models/restaurant_wheel.rb:59]: what's the ampersand for?

File: app/models/restaurant_wheel.rb [hunk 2]
```diff
@@ -30,17 +40,22 @@ def filtered_restaurants
...
+    @total_average_rating ||= filtered_restaurants.sum(&:average_rating).to_d
   end
 
-  def generate_wheel
-    raise 'Must have restuarants to create a wheel!' if ordered_restaurants.empty?
-
-    @wheel = []
-    while @wheel.count < num_stops
-      @wheel += ordered_restaurants
+  def cummulative_weighted_restaurants
```
Comment (LucasSantoro) [inline app/models/restaurant_wheel.rb:47]: nit but only one `m` in cumulative.

File: app/models/restaurant_wheel.rb [hunk 3]
```diff
@@ -30,17 +40,22 @@ def filtered_restaurants
...
+    total = 0
+    @cummulative_weighted_restaurants ||= filtered_restaurants.map do |restaurant|
+      total += restaurant.average_rating / total_average_rating
```
Comment (LucasSantoro) [inline app/models/restaurant_wheel.rb:50]: this isn't actually weighting the averages, right? a 2.5 restaurant with 10 reviews would be given equal space to a 2.5 with none. although, now that i think about it, should we even weight them? and to what degree should a 5-star 10-review spot be given more space than a 5 star with only one review?

File: app/models/restaurant_wheel.rb (hunk 1 above)
Comment (coreyja) [reply app/models/restaurant_wheel.rb:59]: Lonely operator, basically a nil check

File: app/models/restaurant_wheel.rb (hunk 3 above)
Comment (coreyja) [reply app/models/restaurant_wheel.rb:50]: Correct this is weighting the restaurants against each other. It is NOT weighting each restaurants rating based on the number.

If we wanted to do that we should add that to the 'Restaurant#average_rating' method

**Comment by coreyja** [comment]
Merging in case I don't pull this up tonight. I'll fix the typo if I do lol

**Pull request merged**